### ISSUE TITLE
Support for the KDE Plasma desktop

### ISSUE THREAD

**Issue by kveroneau** [drata/drata-agent#22]
I recently started a company that uses Drata, I have yet to receive my Linux laptop for my role, and as I wait, I am naturally going through the various tools and software I will be needing to install and use.  I noticed that Drata is severely limited when it comes to the Linux desktop.  I understand that Linux support in Drata is also fairly new, but I'd also love to help improve it to allow it to be much more diverse when it comes to the Linux desktop.  Namely, I'd love to assist the Drata team with making it compatible with the recent releases of the KDE Plasma desktop, my DE of choice on Linux.  Although, I'd also love to push for pure Debian support as well, since Debian and Ubuntu are generally very compatible, I don't see much friction here.

Before I decide on looking into this feature alone, I thought I'd open an issue here to see if any of the Drata developers have been, or are going to be working on such support for the Drata agent.  I don't believe that it would be too difficult to add, as it will just be a few simple checks in a couple Plasma config files on the system.  Is there currently a system in place within the code base to easily extend it to different DEs?  What would be the recommended way of starting to add support?  I will be issued my Linux laptop in the first week of June I've been told, and so at that time, I will be-able to work more directly with the agent's source code here to enable support for KDE Plasma.

**Comment by dheerajck** [comment]
yeah it sucks that its limited to Gnome

**Comment by ndench** [comment]
#17 appears to say that the agent works _mostly_ fine with KDE.
I'm running it under Arch with i3 and it runs fine. Just can't auto detect encryption, clamav or auto updates. But just upload a screenshot as evidence and it reports my system as compliant.

**Comment by dheerajck** [comment]
But dont they require to update system checks daily ??

I dont know whats the point of drata agent tbh

**Comment by ndench** [comment]
Yes, I just start the drata agent on boot.

Yes, it does need to check daily. They point is to ensure that employees are compliant with security policies across a period of time, rather than just point in time.

For example for SOC Type 1 certification, businesses need to show that their employees computers have a screen lock, anti virus, encryption, etc (among many other things). But they only need to show this for a single point in time.

But for SOC Type 2 certification, the auditor will need proof that the employees computers are compliant over a period of at least 3 months. So the drata agent stores logs which can then be presented to the auditor as proof.

**Comment by dheerajck** [comment]
Drata agent crashes when I was using kde, so I moved to gnome, I think it automatically starts up in the boot, I just set it up and assume I do not need to manually open it everytime I boot up to linux, also I asked whats the point because we are required to upload screenshots manually for some stuff as it cannot detect every requirement

**Comment by ndench** [comment]
You're right, you do not need to open it's GUI, as long as it's running it'll do what it needs to do. 

I agree that it's annoying that we still need to take a bunch of screenshots. 

From what I can see the Drata admin, my computer is still marked as complaint even though it's been a couple weeks since I uploaded screenshots for encryption, antivirus and automatic updates.

If I stop auto booting the agent, then Drata shows a warning that the agent hasn't run. 

I'm unsure how frequently we need to upload more screenshots at this point.

**Comment by NuwanUdara** [comment]
on Kubuntu 24.04 I can start the darata-agent with 
```
drata-agent --no-sandbox
Checking for beta autoupdate feature for deb/rpm distributions
Found package-type: deb
Warning: terminator_CreateInstance: Failed to CreateInstance in ICD 1.  Skipping ICD.

```

The agent does work but I cannot bind it with account. Any workarounds for this?

**Comment by Chevindu** [comment]
> The agent does work but I cannot bind it with account. Any workarounds for this?

@NuwanUdara This worked for me on Kubuntu 24.04: https://github.com/drata/drata-agent/issues/20#issuecomment-2181136642

**Comment by troll-os** [comment]
Installed on Kubuntu 22.04 on my side and the agent, even the UI worked out of the box. 

My guess is that the agent mostly has issues with 24.04, regardless of the DE.

**Comment by rfaurevincent-wiser** [comment]
> But just upload a screenshot as evidence and it reports my system as compliant.

@ndench where do you upload those? the agent runs fine on debian with KDE, but the three checkboxes for disk encryption, AV and unattended upgrades do not get cleared and I don't have see a way to upload the screenshot anywhere.

**Comment by troll-os** [comment]
@rfaurevincent-wiser That would be in the Data web interface. In theory your company should provide you a login and it's there that you post screenshots of your setup

**Comment by rfaurevincent-wiser** [comment]
![image](https://github.com/user-attachments/assets/50309be0-a438-47bd-8ed0-7a264f0aa978)

that's all it shows for now, but I just installed it so maybe on the next sync.

**Comment by troll-os** [comment]
You don't have a "Configure  your devices" panel ? 
I don't remember having to wait to upload proofs in the UI, crosscheck with your IT maybe there's manual process to validate the agent registration ?

**Comment by rfaurevincent-wiser** [comment]
nope, nothing like that. the ability to upload proof screenshots instead of automated checks might be toggled at the company level then, guess I'll have to talk to IT.

**Comment by yekibud** [comment]
I'm on Fedora KDE - has anyone gotten this to work buy converting with alien or building the electron app locally?  Sounds like the app indicator wont show up but maybe a service will do the logging in the background?

**Comment by kaykhan** [comment]
> nope, nothing like that. the ability to upload proof screenshots instead of automated checks might be toggled at the company level then, guess I'll have to talk to IT.

Curious what you ended up doing? i installed ubuntu with LUKS but drata-agent is not recognising that the hard disk is encrypted.

**Comment by maciej-perplexity** [comment]
Hello.

I managed to get drata-agent running on Kubuntu 24.04 by adding this AppArmor profile:
```
$ cat /etc/apparmor.d/drata-agent
abi <abi/4.0>,
include <tunables/global>

profile drata-agent /opt/Drata\ Agent/drata-agent flags=(unconfined) {
  userns,

  # Site-specific additions and overrides. See local/README for details.
  include if exists <local/chrome>
}
```
And running `sudo service apparmor reload`

Now drata-agent runs:
```
$ drata-agent
[196597:0514/152054.175923:ERROR:proxy_config_service_linux.cc(620)] inotify_init failed: Too many open files (24)
[196597:0514/152054.176121:ERROR:file_path_watcher_inotify.cc(315)] inotify_init() failed: Too many open files (24)
[196597:0514/152054.222805:ERROR:proxy_config_service_linux.cc(620)] inotify_init failed: Too many open files (24)
Checking for beta autoupdate feature for deb/rpm distributions
Found package-type: deb
[196658:0514/152058.240300:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 1 times!
[196658:0514/152101.472977:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 2 times!
[196658:0514/152147.367250:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 3 times!
```

But when I try to register my device by going to the link, and then click Open Drata Agent here:

![Image](https://github.com/user-attachments/assets/8a96a173-3e1f-4a8c-b7c9-76ed64e1552f)

Then `auth-drata-agent//token=...` opens _in my default browser_, not in the running drata-agent. How to proceed?

**Comment by gileri** [comment]
>Then auth-drata-agent//token=... opens in my default browser, not in the running drata-agent. How to proceed?

You can catch the link from your browser devtools, then execute drata-agent with the URL as an argument, it should register.

**Comment by maciej-perplexity** [comment]
I tried like this:
```
$ drata-agent https://app.drata.com/agent-auth/[...]?region=NA
```
(I replaced the actual token with [...])

Didn't work, the agent opened as always with "Way to go! The Drata Agent has been installed" followed by instructions on how to register.

**Comment by gileri** [comment]
Looks like drata-agent needs to be running before you can execute the "MIME type action" with the URL :

```shell
killall drata-agent
drata-agent --no-sandbox &
drata-agent --no-sandbox 'auth-drata-agent//token=[...]&region=NA'
```

**Comment by maciej-perplexity** [comment]
Okay, this didn't solve the problem, but I found what caused it.

The reason it didn't work was because I needed a `:` between `auth-drata-agent` and `//token`. I suppose it was removed by how the browser sanitizes a URL with unhandled scheme?

So, the command should be:
```
drata-agent --no-sandbox 'auth-drata-agent://token=[...]&region=NA'
```

Now it registered but is unable to detect that my disk is encrypted, even though it is?

![Image](https://github.com/user-attachments/assets/07011e68-65e1-4dbe-be75-3a4711cfe277)

![Image](https://github.com/user-attachments/assets/485eb194-0bdb-4c99-afe5-237cf1c1f6ee)

The other disk has a Windows 11 installation (with bitlocker)
